File: solutions/4-data-structures-and-libraries/3171/3171-dfs.py
n, l = [int(i) for i in input().split()]

g = [[] for i in range(n)]

for i in range(l):
    x, y = [int(j) - 1 for j in input().split()]
    g[x] += [y]
    g[y] += [x]

visited = [False] * n

def dfs(u):
    if visited[u]: return
    visited[u] = True
    for v in g[u]:
        dfs(v)

dfs(0)

print('COMPLETO' if sum(visited) == n else 'INCOMPLETO')

File: solutions/4-data-structures-and-libraries/3171/3171-dfs.cpp
#include <bits/stdc++.h>

using namespace std;

const int maxn = 100; 
vector<int> g[maxn];
bitset<maxn> visited;

void dfs(int u) {
    if (visited[u]) return;
    visited[u] = true;
    for (int v : g[u]) {
        dfs(v);
    }
}

int main() {
    int n, l;
    cin >> n >> l;
    for (int i = 0; i < l; ++i) {
        int x, y;
        cin >> x >> y;
        --x; --y;
        g[x].push_back(y);
        g[y].push_back(x);
    }
    dfs(0);
    cout << (visited.count() == n ? "COMPLETO" : "INCOMPLETO") << endl;
    return 0;
}
